modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -307,10 +307,26 @@ public void processOpts() {
                 }
             }
         }
+
+        processUsings(objs);
+
         // process enum in models
         return postProcessModelsEnum(objs);
     }
 
+    private void processUsings(Map<String, Object> objs) {
```

Comment:
@mminns 

Sorry, I should have been more clear.

Swagger Codegen allows users to specify their own modifications to templates using the following switches passed to the cli:

```
[(-t <template directory> | --template-dir <template directory>)]
```

So, for example, a user could copy the contents of `modules/swagger-codegen/src/main/resources/csharp` to a local directory such as `/src/csharp-template`. Any of the templates may be modified as long as the template names remain the same and the mustache syntax is valid.

As an example, if I've copied out the templates to `/src/csharp-template`, I can change `api.mustache` to include `System.IO`:

```
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
{{#netStandard}}
using RestSharp.Portable;
{{/netStandard}}
{{^netStandard}}
using RestSharp;
{{/netStandard}}
using {{packageName}}.Client;
{{#hasImport}}using {{packageName}}.{{modelPackage}};
{{/hasImport}}
```

I provided `File` as an example, because it's not uncommon for APIs to have models named as common API inputs/outputs. To demonstrate the conflict with `File`, you'd have to perform this hypothetical adjustment to the usings in your custom template.

However, the problem exists for every model of the existing usings. We had issues with users having conflicts on `Object` with the exiting `using System;`, which lead to the decision to fully qualify all model and infrastructure in the generated C# clients.

Without modifying the templates, you could experience the same conflicts using these not very far-fetched model names:

* Environment
* Type
* Uri
* Version
* A handful of types from RestSharp and RestSharp.Portable.

So, for a better example using the built-in templates, if I had a model called `DateTime` your change would yield:

```
using System;
…
using DateTime = IO.Swagger.Models.DateTime;
``` 

And in this scenario, it would be confusing to the consumers when `DateTime` refers to an aliased type and we're forcing a fully qualified `System.DateTime` anytime someone wants to use a *real* DateTime.

The answer to the above could just be "don't name your models like that". But there may be business needs we're unaware of, or C# could be one of many client SDKs being generated to which enforcing a rule like this could affect many clients of our consumer's generated codebase just to rename a model. It is unobtrusive, however, if every instance in the generated code that needs access to the swagger generated DateTime instance is fully qualified as `IO.Swagger.Models.DateTime` and we don't use aliases as they're done above. Couple that with our inability to know how the templates may have been modified by users, and aliases aren't really viable for us.

I hope that helps. Sorry for the previous short and confusing comment.